__tests__/happy.uploadAttachments.js

```diff
@@ -0,0 +1,107 @@
+import 'dotenv/config';
+
+import onErrorResumeNext from 'on-error-resume-next';
+
+import { DirectLine } from '../src/directLine';
+import { timeouts } from './constants.json';
+import * as createDirectLineOptions from './setup/createDirectLineOptions';
+import fetchAsBase64 from './setup/fetchAsBase64';
+import postActivity from './setup/postActivity';
+import waitForBotEcho from './setup/waitForBotEcho';
+import waitForConnected from './setup/waitForConnected';
+
+describe('Happy path', () => {
+  let unsubscribes;
+
+  beforeEach(() => unsubscribes = []);
+  afterEach(() => unsubscribes.forEach(fn => onErrorResumeNext(fn)));
+
+  describe('upload 2 attachments with text messages', () => {
+    let directLine;
+
+    describe('using REST', () => {
+      beforeEach(() => jest.setTimeout(timeouts.rest));
+
+      test('with secret', async () => {
+        directLine = new DirectLine(await createDirectLineOptions.forREST({ token: false }));
+      });
+
+      test('with token', async () => {
+        directLine = new DirectLine(await createDirectLineOptions.forREST({ token: true }));
+      });
+    });
+
+    // test('using Streaming Extensions', async () => {
+    //   jest.setTimeout(timeouts.webSocket);
+    //   directLine = new DirectLine(await createDirectLineOptions.forStreamingExtensions());
+    // });
+
+    describe('using Web Socket', () => {
+      beforeEach(() => jest.setTimeout(timeouts.webSocket));
+
+      test('with secret', async () => {
+        directLine = new DirectLine(await createDirectLineOptions.forWebSocket({ token: false }));
+      });
+
+      test('with token', async () => {
+        directLine = new DirectLine(await createDirectLineOptions.forWebSocket({ token: false }));
+      });
+    });
+
+    afterEach(async () => {
+      // If directLine object is undefined, that means the test is failing.
+      if (!directLine) { return; }
+
+      unsubscribes.push(directLine.end.bind(directLine));
+      unsubscribes.push(await waitForConnected(directLine));
+
+      const sendingActivity = {
```

Comment:
Variables should be nouns - not verbs. `acitivity` or `message` might be better options.

Reply:
Is `activityInTransit` better?